src/main.cpp

```diff
@@ -241,14 +233,15 @@ Napi::Value DDWAFContext::run(const Napi::CallbackInfo& info) {
     mlog("Set total_runtime");
     res.Set("totalRuntime", Napi::Number::New(env, result.total_runtime));
   }
-  if (code != DDWAF_GOOD) {
+  if (code != DDWAF_OK) {
     res.Set("data", Napi::String::New(env, result.data));
   }
-  if (code == DDWAF_MONITOR) {
-    res.Set("action", Napi::String::New(env, "monitor"));
-  }
-  if (code == DDWAF_BLOCK) {
-    res.Set("action", Napi::String::New(env, "block"));
+  if (code == DDWAF_MATCH) {
+    res.Set("status", Napi::String::New(env, "match"));
+    Napi::Array actions = Napi::Array::New(env, result.actions.size);
+    for (uint32_t i = 0; i < result.actions.size; ++i) {
```

Comment:
`result.actions` has always a value? Can it be `nullptr` and create `SEGFAULT` here?

Reply:
Yes, it looks fine